plutus-contract/test/Spec/TxConstraints/RequiredSigner.hs

```diff
@@ -0,0 +1,217 @@
+{-# LANGUAGE TemplateHaskell     #-}
+{-# LANGUAGE TypeApplications    #-}
+{-# LANGUAGE TypeFamilies        #-}
+module Spec.TxConstraints.RequiredSigner(tests) where
+
+import Control.Monad (void)
+import Data.Map qualified as Map
+import Data.Void (Void)
+import Test.Tasty (TestTree, testGroup)
+
+import Data.Maybe (fromJust)
+import Data.String (fromString)
+import Ledger qualified
+import Ledger.Ada qualified as Ada
+import Ledger.CardanoWallet as CW
+import Ledger.Constraints.OffChain qualified as Constraints (paymentPubKey, plutusV1OtherScript,
+                                                             plutusV1TypedValidatorLookups, unspentOutputs)
+import Ledger.Constraints.OnChain.V1 qualified as Constraints
+import Ledger.Constraints.TxConstraints qualified as Constraints (mustBeSignedBy, mustIncludeDatum, mustPayToTheScript,
+                                                                  mustSpendScriptOutput, requiredSignatories)
+import Ledger.Tx qualified as Tx
+import Ledger.Typed.Scripts qualified as Scripts
+import Plutus.Contract as Con
+import Plutus.Contract.Test (assertFailedTransaction, assertValidatedTransactionCount, checkPredicateOptions,
+                             defaultCheckOptions, mockWalletPaymentPubKey, mockWalletPaymentPubKeyHash, w1, w2)
+import Plutus.Trace qualified as Trace
+import Plutus.V1.Ledger.Api (Redeemer (Redeemer), ScriptContext (scriptContextTxInfo), ToData (toBuiltinData), TxInfo)
+import Plutus.V1.Ledger.Contexts (txSignedBy)
+import Plutus.V1.Ledger.Scripts (ScriptError (EvaluationError), unitDatum)
+import PlutusTx qualified
+import PlutusTx.Prelude qualified as P
+import Prelude
+import Wallet.Emulator.Wallet (signPrivateKeys, walletToMockWallet)
+
+tests :: TestTree
+tests =
+    testGroup "required signer"
+        [
+        -- tests that use txSignedBy on-chain
+          ownWallet
+        , otherWallet
+        , missingRequiredSignature
+        , otherWalletNoSigningProcess
+        , withoutOffChainMustBeSignedBy
+        -- tests that use mustBeSignedBy constraint on-chain
+        , ownWalletOnChainConstraint
+        , missingRequiredSignatureOnChainConstraint
+        , withoutOffChainMustBeSignedByOnChainConstraint
+        ]
+
+mustBeSignedByContract :: Ledger.PaymentPubKey -> Ledger.PaymentPubKeyHash -> Scripts.TypedValidator UnitTest -> Contract () Empty ContractError ()
+mustBeSignedByContract pk pkh tv = do
+    let validator = Scripts.validatorScript tv
+        lookups1 = Constraints.plutusV1TypedValidatorLookups tv
+        tx1 = Constraints.mustPayToTheScript () (Ada.lovelaceValueOf 25000000)
+    ledgerTx1 <- submitTxConstraintsWith lookups1 tx1
+    awaitTxConfirmed $ Tx.getCardanoTxId ledgerTx1
+
+    utxos <- utxosAt (Ledger.scriptHashAddress $ Scripts.validatorHash tv)
+    let orefs = fst <$> Map.toList utxos
+        redeemer = Redeemer $ toBuiltinData pkh
+        lookups2 =
+            Constraints.plutusV1OtherScript validator
```

Comment:
Since you're using TypedValidator for your contract, I suggest using `Constraints.plutusV1TypedValidatorLookups`. Which implies that you can use `collectFromTheScript` later on